File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
      let maxProf = 0;
  let buy = 0;
  let sell = buy + 1;

  while (sell < prices.length) {
    if (prices[sell] < prices[buy]) {
      buy = sell;
    }
    maxProf = Math.max(maxProf, prices[sell] - prices[buy]);
    sell++;
  }

  return maxProf;

};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        buy = 0
        sell = buy + 1
        while sell < len(prices):
            if prices[sell] < prices[buy]:
                buy = sell
            elif prices[sell] > prices[buy]:
                profit = max(prices[sell]-prices[buy],profit)
            sell = sell + 1
        return profit
        